pkgs/build-support/xen/default.nix

```
@@ -114,10 +115,14 @@ let
     util-linux.bin
     which
   ];
+
+  genericPatches = [ "0001-tools-libxl-search-PATH-for-QEMU-if-QEMU_XEN_PATH-is.patch" ];
 in
 
 stdenv.mkDerivation (finalAttrs: {
-  inherit pname version patches;
+  inherit pname version;
+
+  patches = optionals withGenericPatches genericPatches ++ patches;
```

```suggestion
  patches = [ (fetchpatch {...}) ] ++ patches;
```

#406638 will be deleting the second function call after the branchoff, so we might as well not add things to it.